File: examples/maven-java/src/main/java/com/example/orgchart/Address.java
package com.example.orgchart;

import com.fasterxml.jackson.annotation.JsonClassDescription;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;

/**
 * A postal address nested inside {@link ContactInfo}.
 */
@JsonClassDescription("A postal address.")
public record Address(
        @JsonPropertyDescription("Street and house number")
        String street,
        @JsonPropertyDescription("City or town")
        String city,
        @JsonPropertyDescription("Country")
        String country,
        @JsonPropertyDescription("Postal code")
        String zipCode) {
}

File: ksp-integration-tests/src/main/kotlin/me/kpavlov/kt/schema/integration/type/Address.kt
package me.kpavlov.kt.schema.integration.type

import me.kpavlov.kt.schema.Description
import me.kpavlov.kt.schema.Schema

/**
 * A postal address for deliveries and billing.
 *
 * @author Konstantin Pavlov
 * @see [String]
 * @since 1.0
 * @param street This `@param` description should be ignored
 * @property street This `@property` description should be ignored
 */
@Schema
data class Address(
    @Description("Street address, including house number")
    val street: String,
    @Description("City or town name")
    val city: String,
    @Description("Postal or ZIP code")
    val zipCode: String,
    @Description("Two-letter ISO country code; defaults to US")
    val country: String = "US",
)
